cli/src/main/kotlin/commands/UploadResultToSw360Command.kt

```diff
@@ -0,0 +1,142 @@
+    epilog = "EXPERIMENTAL: The command is still in development and usage will likely change in the near future. The " +
+            "command expects that a Sw360Storage for the scanner is configured."
+) {
+    private val ortFile by option(
+        "--ort-file", "-i",
+        help = "The ORT result file to read as input."
+    ).convert { it.expandTilde() }
+        .file(mustExist = true, canBeFile = true, canBeDir = false, mustBeWritable = false, mustBeReadable = true)
+        .convert { it.absoluteFile.normalize() }
+        .required()
+        .inputGroup()
+
+    private val globalOptionsForSubcommands by requireObject<GlobalOptions>()
+
+    override fun run() {
+        val (ortResult, duration) = measureTimedValue { ortFile.readValue<OrtResult>() }
+
+        log.perf {
+            "Read ORT result from '${ortFile.name}' (${ortFile.formatSizeInMib}) in ${duration.inMilliseconds}ms."
+        }
+
+        val sw360Config = globalOptionsForSubcommands.config.scanner?.storages?.values
+            ?.filterIsInstance<Sw360StorageConfiguration>()?.singleOrNull()
+
+        requireNotNull(sw360Config) {
+            "No SW360 storage is configured for the scanner."
+        }
+
+        createSw360Connection(sw360Config)?.let { sw360Connection ->
+            val sw360ReleaseClient = sw360Connection.releaseAdapter
+
+            ortResult.getPackages(omitExcluded = true).map { (pkg, _) ->
+                try {
+                    val release = SW360Release()
+                        .setMainLicenseIds(pkg.declaredLicenses)
+                        .setName(pkg.id.name)
+                        .setVersion(pkg.id.version)
+
+                    sw360ReleaseClient.createRelease(release)?.let {
+                        log.debug { "Release ${release.name}-${release.version} created in SW360." }
+                    }
+                } catch (e: SW360ClientException) {
+                    log.error { "Could not create the release in SW360 for the reason: ${e.message}" }
+                }
+            }
+        }
+    }
+
+    /**
+     * Crates a [SW360Connection] based on the given [Sw360StorageConfiguration].
+     */
+    private fun createSw360Connection(config: Sw360StorageConfiguration): SW360Connection? {
+        require(config.restUrl.isNotBlank()) {
+            "URL for SW360 REST API is missing."
+        }
+
+        require(config.authUrl.isNotBlank()) {
+            "URL for SW360 authentication server is missing."
+        }
+
+        require(config.username.isNotBlank()) {
+            "Username for SW360 is missing."
+        }
+
+        require(config.password.isNotBlank()) {
+            "Password of the username for SW360 is missing."
+        }
+
+        require(config.clientId.isNotBlank()) {
+            "Client ID of the SW360 instance is missing."
+        }
+
+        require(config.clientPassword.isNotBlank()) {
+            "Client password of the SW360 instance is missing."
+        }
```

Comment:
These checks should IMO go into a constructor of `Sw360StorageConfiguration`, so they are performed no matter how the config is created / where it is used.

Reply:
The checks are also done by _SW360ClientConfig_; so if you rely on those, yo do not have to repeat them here.

Reply:
@sschuberth: This is also not done by the other `ScanStorageConfiguration` implemenations. Probably this should also be done by the other implemenations??

@oheger-bosch: Thanks for the hint. So i can only use the `SW360ClientConfig::createConfig` without checking the parameters twice?

Reply:
> This is also not done by the other `ScanStorageConfiguration` implemenations.

Which is not a reason for you not to do it, esp. as you already have written the code 😉 You should aim to better than other implementations, not only just as good.

> Probably this should also be done by the other implemenations??

True, probably yes, but that's out of scope for *your* change here 😉 

Anyway, now that we know this is done internally by `SW360ClientConfig`, we should rely on those checks and not duplicate them, like @oheger-bosch suggested.